File: leecode/201-300/232-用栈实现队列.cpp
class MyQueue {
public:
    /** Initialize your data structure here. */
    stack<int> ru;
    stack<int> chu;
    MyQueue() {
    }
    /** Push element x to the back of queue. */
    void push(int x) {
        ru.push(x);
    }
    /** Removes the element from in front of queue and returns that element. */
    int pop() {
        int x;
        if (chu.empty()) {
            while (!ru.empty()) {
                int x = ru.top(); ru.pop();
                chu.push(x);
            }
        }
        x = chu.top(); chu.pop();
        return x;
    }
    
    /** Get the front element. */
    int peek() {
        if (chu.empty()) {
            while (!ru.empty()) {
                int x = ru.top(); ru.pop();
                chu.push(x);
            }
        }
        return chu.top();
    }
    
    /** Returns whether the queue is empty. */
    bool empty() {
        if (ru.empty() && chu.empty())    
            return true;
        return false;
    }
};

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue* obj = new MyQueue();
 * obj->push(x);
 * int param_2 = obj->pop();
 * int param_3 = obj->peek();
 * bool param_4 = obj->empty();
 */

File: leecode/1-500/201-300/232-用栈实现队列.py
class MyQueue:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stacka = []
        self.stackb = []

    def push(self, x: int) -> None:
        """
        Push element x to the back of queue.
        """
        self.stacka.append(x)

    def pop(self) -> int:
        """
        Removes the element from in front of queue and returns that element.
        """
        if not self.stackb:
            while self.stacka:
                self.stackb.append(self.stacka.pop())
        return self.stackb.pop()

    def peek(self) -> int:
        """
        Get the front element.
        """
        if not self.stackb:
            while self.stacka:
                self.stackb.append(self.stacka.pop())
        return self.stackb[-1]


    def empty(self) -> bool:
        """
        Returns whether the queue is empty.
        """
        if not self.stacka and not self.stackb:
            return True
        else:
            return False

# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()